ccm_web/package.json

```diff
@@ -5,7 +5,8 @@
   "scripts": {
     "dev:server": "nodemon --inspect=0.0.0.0:9229 --config nmon_config.json server/src/main.ts",
     "dev:client": "webpack --watch --config webpack/webpack.dev.ts",
-    "dev": "npm run dev:server & npm run dev:client",
+    "predev": "wait-port ccm_db:3306",
```

Comment:
I did not shop around very much, But I tried what worked. 
For Running code locally (With out type compilation way using `ts-node`), the docker-compose command `npm run dev` is a run time step. So this worked well.

For Prod mode running locally, we have `preprod` step in `package.json` this is a compilation step and we don't want to wait on DB container start up at all since for prod it is always will be running. 
So I couldn't figure out how this to work it in prod. This `wait-port ccm_db_prod:3306` should go in `docker-compose-prod.yml`. I tried to fit in this with `command` variable but did not work. So for running Prod mode Locally waiting for DB is still an issue. So I decide could be tackled later.

The most popular seems to be [wait-for](https://github.com/eficode/wait-for). There are other method for doing via [dockerfile](https://docs.docker.com/compose/startup-order/) as well some of then required shell file to do it, which typescript seems to consuming during compilation step and not putting in the target directory. So i don't know what is the best way to do this.

Reply:
In `docker-compose-prod.yml` we can do something like this 

`command: ["wait-port", "ccm_web_prod:5432", "--", "node", "server migrator up","--" "node", "server/src/main.js",]`

But this got bypassed what is in the `Dockerfile` `CMD` and I did not wanted wait-port in dockerfiles. 

So I felt this issue can be taken as separate issue.

Reply:
This comment might be relevant here too: https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/80#discussion_r636207358

What you're doing now seems okay to me, I just was curious what you found.